test/RelayHub.test.ts

```diff
@@ -1574,7 +1577,34 @@ contract('RelayHub', function ([_, relayOwner, relayManager, relayWorker, incorr
     const gas = 4e6
 
     beforeEach(async function () {
-      relayHubInstance = await deployHub(penalizerMock)
+      relayHubInstance = await deployHub()
+    })
+
+    context('smart contract setter', function () {
+      it('should start out with an unset address', async function () {
+        assert.equal(await relayHubInstance.penalizer(), constants.ZERO_ADDRESS, 'hub penalizer does not match zero address')
+      })
+
+      it('should not be able to set penalizer if not owner', async function () {
+        await expectRevert(
+          relayHubInstance.setPenalizer(penalizerMock, { from: other }), 'caller is not the owner'
+        )
+      })
+
+      it('should be able to set penalizer if owner', async function () {
+        await relayHubInstance.setPenalizer(penalizerMock, { from: await relayHubInstance.owner() })
+        assert.equal(await relayHubInstance.penalizer(), penalizerMock, 'hub penalizer value not set as expected')
+      })
+
+      it('should not be able to set penalizer if already set', async function () {
+        // set penalizer the first time
+        await relayHubInstance.setPenalizer(penalizerMock, { from: await relayHubInstance.owner() })
+        // attempt to set a second time
+        await expectRevert(
+          relayHubInstance.setPenalizer(other, { from: await relayHubInstance.owner() }), 'penalizer already set'
```

Comment:
setting the `owner` field isn't strictly needed here, as truffle will fetch it on its own. 

just wanted to have this as explicit as possible since it is what is being tested